### PR TITLE
Add support for nacks to the Kafka connector

### LINKED ISSUE #25969: Add tests for MP RM 3.0 Nack with kafka connector
In MP RM 2.0 Negative Acknowledges (Nacks) were added in case of error cases

We should add some test cases similar to the existing Acknowledgement test cases 

For MP RM

### PR DESCRIPTION
**Pull request by Azquelt**
Outgoing Kafka connector will nack messages it cannot deliver.

Incoming Kafka connector will shut down and log an error if one of the messages it created is nacked.

Fixes #25969

### CODE REVIEW

**Comment by Azquelt** [comment]
#build
#libby

**Comment by abutch3r** [comment]
Overall looks good, need to investigate the test failures for common and the JSON-b tests which for some reason are now impacted by kafka

**Comment by Azquelt** [comment]
The Json-B tests are likely impacted somehow by either the refactoring of the framework classes into a common package, or by the changes needed to common classes to accommodate testing the 3.0 API.

**Comment by Azquelt** [comment]
`AbstractDeliveryBean` was updated so that it could report nacks so it now uses the new class `TestMessageImpl`.

I'd changed `JsonbTest` to include the whole package instead of just the two classes it needed, but hadn't realised that pulls in a bean that refers to Kafka classes which causes a class not found exception during CDI startup.

I should have split up the framework classes which need Kafka from those that don't. We should also probably consolidate some of the common app setup code into a utility method. We can change this in a future PR.

**Comment by Azquelt** [comment]
#build
#libby

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops [hunk 1]
```diff
@@ -82,6 +82,11 @@ kafka.create.outgoing.retry.CWMRX1010W=CWMRX1010W: A Kafka outgoing connector fo
 kafka.create.outgoing.retry.CWMRX1010W.explanation=An error occurred when the Kafka connector was set up for an outgoing reactive messaging channel. This might be caused by a problem with the connector configuration, or by a problem resolving the Kafka broker hostname.
 kafka.create.outgoing.retry.CWMRX1010W.useraction=Review the error message, the messages.log file on the server, and FFDC logs to identify whether the problem was transient and was resolved by retrying the connector setup.
 
+# {0} = kafka record data, {1} = exception details
+# 'nack' is the name of a method and should not be translated 
+kafka.input.message.nacked.CWMRX1011E=CWMRX1011E: The nack method was called on a message created from a Kafka record, indicating that it was not be processed successfully. The reactive message stream will be shut down. The record was: {0}. The exception was: {1}.
```
Comment (posmith) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops:87]: Extraneous word, 'be', in the statement, "indicating that it was not be processed successfully".  Please remove.

Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops:87]:
```suggestion
kafka.input.message.nacked.CWMRX1011E=CWMRX1011E: The nack method was called on a message that was created from a Kafka record, which indicates that the message was not processed successfully. The reactive message stream is shut down. The record was: {0}. The exception was: {1}.
```

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops [hunk 2]
```diff
@@ -82,6 +82,11 @@ kafka.create.outgoing.retry.CWMRX1010W=CWMRX1010W: A Kafka outgoing connector fo
...
+kafka.input.message.nacked.CWMRX1011E.explanation=A message sent by the kafka connector caused a failure somewhere in the application which resulted in the message being negatively acknowledged. As this message has not been successfully processed, the committed offset will not be advanced for the partition that the message came from and so no more messages can be processed from this partition.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops:88]:
```suggestion
kafka.input.message.nacked.CWMRX1011E.explanation=A message that was sent by the Kafka connector caused a failure somewhere in the application, which resulted in the message being negatively acknowledged. As this message was not successfully processed, the committed offset is not advanced for the partition that the message came from. As a result, no more messages can be processed from this partition.
```

File: dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops [hunk 3]
```diff
@@ -82,6 +82,11 @@ kafka.create.outgoing.retry.CWMRX1010W=CWMRX1010W: A Kafka outgoing connector fo
...
+kafka.input.message.nacked.CWMRX1011E.useraction=Review the error message and the messages.log file on the server to identify the problem. If stopping processing of messages is not the desired behavior when message processing fails, add logic in your application which provides alternative handling of messages which are negatively-acknowledged.
```
Comment (dmuelle) [inline dev/com.ibm.ws.microprofile.reactive.messaging.kafka/resources/com/ibm/ws/microprofile/reactive/messaging/kafka/resources/ReactiveMessaging.nlsprops:89]:
```suggestion
kafka.input.message.nacked.CWMRX1011E.useraction=Review the error message and the messages.log file on the server to identify the problem. If you want to continue to process more messages from a partition after a message processing failure, add logic in your application that provides alternative handling of messages that are negatively-acknowledged.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, let me know if you have any questions or if I've misinterpreted the meaning. Thanks

**Comment by Azquelt** [comment]
The code changes were merged under #26385.

The ID comments will be addressed under #26428

**Pull request closed without merging**